lib/collection/header-list.js

```diff
@@ -30,11 +30,22 @@ _.assign(HeaderList.prototype, /** @lends HeaderList.prototype */ {
      * @returns {Number}
      */
     contentSize: function () {
-        if (!this.count()) { return 0; }
+        var raw = E,
+
+            headers = this.reduce(function (acc, header) {
```

Comment:
Why can't you append to the string `raw` directly?

Reply:
It will be like this:
```javascript
raw = this.reduce(..).join(CRLF);
raw !== E && (raw += CRLF);
```

unparsing headers separately looks cleaner, but the alternative I can think of is one shared above.